lib/dpkg/tarfn.c

```diff
@@ -485,6 +488,8 @@ tar_extractor(struct tar_archive *tar)
 		}
 
 		if (h.name[0] == '\0') {
+			if(tar->err.str != NULL)
+				free(tar->err.str);
 			status = dpkg_put_error(&tar->err,
```

Comment:
This didn't make sense. I don't see any code paths were tar->err is set and the loop continues. If there's one such occurrence then that needs to stop the loop, not continue and overwrite the error variable. If you have found one such occurrence I'd appreciate a description of how/where that happens.

Reply:
https://github.com/guillemj/dpkg/blob/cf8fe0d6e74fdc6970ceb17e44ec9f67901120b8/lib/dpkg/tarfn.c#L337

I have to admit - my fix was incorrect. The correct fix would be to fix that line.